lib.go

```diff
@@ -212,6 +228,12 @@ func (vm *VM) Query(
 		return nil, gasUsed, err
 	}
 
+	gasForDeserialization := uint64(deserCost.Mul(uint64(len(data))).Int())
+	if gasLimit < gasForDeserialization+gasUsed {
+		return nil, gasUsed, fmt.Errorf("gas depletion")
```

Comment:
Let's try something a bit more helpful

```suggestion
		return nil, gasUsed, fmt.Errorf("Insufficient gas left to deserialize contract execution result (%d bytes)", len(data))
```